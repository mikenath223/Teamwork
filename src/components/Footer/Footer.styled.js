import styled from 'styled-components';

export const Wrapper = styled.footer`
  background-color: ${({ theme }) => theme.colors.pink};
  margin-top: 70px;
`;

export const GroupText = styled.h5`
  font-weight: ${({ theme }) => theme.fontWeight.bold};
  font-size: ${({ theme }) => theme.textSizes.regular};
  color: ${({ theme }) => theme.colors.white};
  width: max-content;
  text-transform: uppercase;
  letter-spacing: 2px;
  margin-bottom: 10px;
`;

export const LinkText = styled.a`
  font-size: ${({ theme }) => theme.textSizes.small};
  color: ${({ theme }) => theme.colors.white};
  display: block;
  text-transform: capitalize;
  margin: 10px 0;
  cursor: pointer;
  width: max-content;
`

export const BottomBar = styled.div`
  display: flex;
  align-items: center; 
  justify-content: space-between;
  background-color: ${({ theme }) => theme.colors.darkpink};
  margin-top: 40px;
  padding: 0 50px;
  height: 40px;

  img {
    width: 120px;
    height: 20px;
  }
`;

export const CopyrightText = styled.h6`
  font-size: ${({ theme }) => theme.textSizes.small};
  color: ${({ theme }) => theme.colors.white};
  margin: 0;
`