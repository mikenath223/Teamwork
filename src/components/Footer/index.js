import React from 'react';
import { Row, Col } from 'react-bootstrap';
import SocialIcons from 'assets/socialicons.png';
import {
  Wrapper,
  GroupText,
  LinkText,
  BottomBar,
  CopyrightText
} from './Footer.styled';

const Footer = () => (
  <Wrapper>
    <Row className="justify-content-around mx-5">
      <Col className="pl-5 mt-5">
        <GroupText>Company</GroupText>
        <LinkText>About Teamwork</LinkText>
        <LinkText>Careers</LinkText>
        <LinkText>Customers</LinkText>
        <LinkText>Security</LinkText>
        <LinkText>News</LinkText>
        <LinkText>Brand</LinkText>
        <LinkText>Contact us</LinkText>
      </Col>
      <Col className="pl-5 mt-5">
        <GroupText>Products</GroupText>
        <LinkText>Teamwork</LinkText>
        <LinkText>Desk</LinkText>
        <LinkText>Chat</LinkText>
        <LinkText>Spaces</LinkText>
        <LinkText>CRM</LinkText>
        <LinkText>View all products</LinkText>
      </Col>
      <Col className="pl-5 mt-5">
        <GroupText>Solutions</GroupText>
        <LinkText>Marketing agency</LinkText>
        <LinkText>Marketing teams</LinkText>
        <LinkText>Product teams</LinkText>
        <LinkText>Support teams</LinkText>
        <LinkText>Professional services</LinkText>
        <LinkText>Enterprises</LinkText>
      </Col>
      <Col className="pl-5 mt-5">
        <GroupText>Resources</GroupText>
        <LinkText>Blog</LinkText>
        <LinkText>Webinars</LinkText>
        <LinkText>Ebooks</LinkText>
        <LinkText>Guides</LinkText>
        <LinkText>Templates</LinkText>
        <LinkText>Project management guide</LinkText>
        <LinkText>View all resources</LinkText>
      </Col>
      <Col className="pl-5 mt-5">
        <GroupText>Support</GroupText>
        <LinkText>Support Center</LinkText>
        <LinkText>Training</LinkText>
        <LinkText>API</LinkText>
        <LinkText>Integrations</LinkText>
        <LinkText>Partners</LinkText>
        <LinkText>Startups</LinkText>
        <LinkText>Roadmap</LinkText>
        <LinkText>Status</LinkText>
      </Col>
    </Row>
    <BottomBar>
      <CopyrightText>
        Â© 2020 Teamwork.com, Ltd. All rights reserved.
      </CopyrightText>
      <img src={SocialIcons} alt="" />
    </BottomBar>
  </Wrapper>
);

export default Footer;