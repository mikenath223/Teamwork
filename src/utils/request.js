import { auth, generateUserDocument, fire } from './firebaseConfig';
import db from './firebaseConfig';

/*
/ Util function to signup new user 
/ with email and password
*/
export const createUserWithEmailAndPassword = async (email, password, displayName, setError) => {
  try {
    const { user = {} } = await auth.createUserWithEmailAndPassword(email, password)
    await generateUserDocument(user, { displayName });
  } catch (error) {
    setError(error.message);
  }
};

/*
/ Util function to signIn user 
/ with email and password
*/
export const signInUserWithEmailAndPassword = (email, password, setError) => {
  auth.signInWithEmailAndPassword(email, password).catch(() => {
    setError("Error signing in with password and email!");
  });
}

/*
/ Util function to reset user password
*/
export const resetUserPassword = (email, setEmailHasBeenSent, setError) => {
  auth.sendPasswordResetEmail(email)
    .then(() => {
      setEmailHasBeenSent(true);
    })
    .catch(() => {
      setError("Error resetting password");
    });
}

/*
/ Util function to get users
*/
export const getUsers = async () => {
  try {
    let querySnapshot = [];
    await db.collection('users').get().then(query =>
      query.forEach(doc => querySnapshot.push({ id: doc.id, data: doc.data() })))
    return querySnapshot
  } catch (error) {
    console.log('Failed to retrieve users', JSON.stringify(error));
    throw new Error(error);
  }
};

/*
/ Util function to get chats
*/
export const getChats = async (recieverId, currentUserId, runIt) => {
  try {
    await new Promise(function (resolve) {
      return db.collection("chat").doc(recieverId)
        .onSnapshot(function (doc) {
          const chats = doc.data();
          if (chats.data) {
            const filteredChats = chats.data.filter(item =>
              item.recieverId === currentUserId);
            resolve(runIt(filteredChats));
          };
        });
    })
  } catch (error) {
    console.log('Chat uplink ended', JSON.stringify(error));
    throw new Error(error);
  }
}

/*
/ Util function to send chats
*/
export const sendFirestoreChat = (message, senderId, recieverId) => {
  try {
    db.collection('chat').doc(senderId).get().then((doc) => {
      if (doc.exists) {
        db.collection('chat').doc(senderId).update({
          data: fire.firestore.FieldValue.arrayUnion({
            message, recieverId, date: new Date().toLocaleString()
          })
        });
      } else {
        db.collection('chat').doc(senderId).set({
          data: [{
            message, recieverId, date: new Date().toLocaleString()
          }]
        });
      }
    })
  } catch (error) {
    console.log('Failed to send chat', JSON.stringify(error));
    throw new Error(error);
  }
};

/*
/ Util function to group chats
*/
export const runDispatch = (moment, dispatch, setChats,
  chats = [], count = 0) => (input) => {

    if (count >= 2) {
      count = 0;
      chats = [];
    }
    chats.push(...input);
    if (chats.length) {
      chats = chats.sort((a, b) =>
        new Date(a.date) - new Date(b.date))
      chats = chats.map(item => ({
        date: moment(item.date).fromNow(),
        message: item.message,
        recieverId: item.recieverId
      }));
      dispatch(setChats(chats));
    };
    count += 1;
  };

/*
/ Util function to save meetings
*/
export const sendFirebaseEvent = (event, currentUserId) => {
  db.collection('meetings').doc(currentUserId).get().then((doc) => {
    if (doc.exists) {
      db.collection('meetings').doc(currentUserId).update({
        data: fire.firestore.FieldValue.arrayUnion(event)
      });
    } else {
      db.collection('meetings').doc(currentUserId).set({
        data: [event]
      });
    }
  })
}