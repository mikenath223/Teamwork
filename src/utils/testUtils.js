import React from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/global';
import { render } from '@testing-library/react';


export const withThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);