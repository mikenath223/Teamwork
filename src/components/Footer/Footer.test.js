import React from 'react';
import 'jest-styled-components';
import { withThemeProvider } from 'utils/testUtils';
import Footer from './index';

describe('Footer', function () {
  it('should match snapshot', () => {
    const { container } = withThemeProvider(<Footer />)

    expect(container).toMatchSnapshot();
  })
})