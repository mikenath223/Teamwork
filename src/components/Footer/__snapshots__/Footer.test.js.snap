// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
.c0 {
  background-color: #FF4B34;
  margin-top: 70px;
}

.c1 {
  font-weight: 700;
  font-size: 14px;
  color: #fff;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  text-transform: uppercase;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  margin-bottom: 10px;
}

.c2 {
  font-size: 12px;
  color: #fff;
  display: block;
  text-transform: capitalize;
  margin: 10px 0;
  cursor: pointer;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #D73520;
  margin-top: 40px;
  padding: 0 50px;
  height: 40px;
}

.c3 img {
  width: 120px;
  height: 20px;
}

.c4 {
  font-size: 12px;
  color: #fff;
  margin: 0;
}

<div>
  <footer
    class="c0"
  >
    <div
      class="justify-content-around mx-5 row"
    >
      <div
        class="pl-5 mt-5 col"
      >
        <h5
          class="c1"
        >
          Company
        </h5>
        <a
          class="c2"
        >
          About Teamwork
        </a>
        <a
          class="c2"
        >
          Careers
        </a>
        <a
          class="c2"
        >
          Customers
        </a>
        <a
          class="c2"
        >
          Security
        </a>
        <a
          class="c2"
        >
          News
        </a>
        <a
          class="c2"
        >
          Brand
        </a>
        <a
          class="c2"
        >
          Contact us
        </a>
      </div>
      <div
        class="pl-5 mt-5 col"
      >
        <h5
          class="c1"
        >
          Products
        </h5>
        <a
          class="c2"
        >
          Teamwork
        </a>
        <a
          class="c2"
        >
          Desk
        </a>
        <a
          class="c2"
        >
          Chat
        </a>
        <a
          class="c2"
        >
          Spaces
        </a>
        <a
          class="c2"
        >
          CRM
        </a>
        <a
          class="c2"
        >
          View all products
        </a>
      </div>
      <div
        class="pl-5 mt-5 col"
      >
        <h5
          class="c1"
        >
          Solutions
        </h5>
        <a
          class="c2"
        >
          Marketing agency
        </a>
        <a
          class="c2"
        >
          Marketing teams
        </a>
        <a
          class="c2"
        >
          Product teams
        </a>
        <a
          class="c2"
        >
          Support teams
        </a>
        <a
          class="c2"
        >
          Professional services
        </a>
        <a
          class="c2"
        >
          Enterprises
        </a>
      </div>
      <div
        class="pl-5 mt-5 col"
      >
        <h5
          class="c1"
        >
          Resources
        </h5>
        <a
          class="c2"
        >
          Blog
        </a>
        <a
          class="c2"
        >
          Webinars
        </a>
        <a
          class="c2"
        >
          Ebooks
        </a>
        <a
          class="c2"
        >
          Guides
        </a>
        <a
          class="c2"
        >
          Templates
        </a>
        <a
          class="c2"
        >
          Project management guide
        </a>
        <a
          class="c2"
        >
          View all resources
        </a>
      </div>
      <div
        class="pl-5 mt-5 col"
      >
        <h5
          class="c1"
        >
          Support
        </h5>
        <a
          class="c2"
        >
          Support Center
        </a>
        <a
          class="c2"
        >
          Training
        </a>
        <a
          class="c2"
        >
          API
        </a>
        <a
          class="c2"
        >
          Integrations
        </a>
        <a
          class="c2"
        >
          Partners
        </a>
        <a
          class="c2"
        >
          Startups
        </a>
        <a
          class="c2"
        >
          Roadmap
        </a>
        <a
          class="c2"
        >
          Status
        </a>
      </div>
    </div>
    <div
      class="c3"
    >
      <h6
        class="c4"
      >
        Â© 2020 Teamwork.com, Ltd. All rights reserved.
      </h6>
      <img
        alt=""
        src="socialicons.png"
      />
    </div>
  </footer>
</div>
`;
